lib/matplotlib/ticker.py

```diff
@@ -2380,21 +2383,26 @@ def nonsingular(self, vmin, vmax):
             return 1, 10  # initial range, no data plotted yet
 
         if vmin > vmax:
+            swap = True
             vmin, vmax = vmax, vmin
+        else:
+            swap = False
         if vmax <= 0:
             cbook._warn_external(
                 "Data has no positive values, and therefore cannot be "
                 "log-scaled.")
-            return 1, 10
-
-        minpos = self.axis.get_minpos()
-        if not np.isfinite(minpos):
-            minpos = 1e-300  # This should never take effect.
-        if vmin <= 0:
-            vmin = minpos
-        if vmin == vmax:
-            vmin = _decade_less(vmin, self._base)
-            vmax = _decade_greater(vmax, self._base)
+            vmin, vmax = 1, 10
```

Comment:
Is it  intended that this might be swapped depending on `swap`?

Reply:
Yes (at least it's consistent with the base Locator.nonsingular).